src/main/java/me/zeroeightsix/kami/module/modules/movement/ElytraFlight.kt

```diff
@@ -404,4 +548,16 @@ class ElytraFlight : Module() {
     init {
         defaultSetting.settingListener = SettingListeners { if (defaultSetting.value) defaults() }
     }
+
+    /* Reset isFlying state when switch mode */
+    init {
+        mode.settingListener = SettingListeners {
+            if (mode.value != previousMode) {
+                mc.player.capabilities.isFlying = false
+                isFlying = false
+                isPacketFlying = false
+            }
+            previousMode = mode.value
```

Comment:
now that this is a setting listener, there is 0 reason to have previousMode anymore